Marketo Static Lists documentation (#1007)
rudderlabs/rudderstack-docs: docs/destinations/streaming-destinations/marketo-static-lists.mdx

| :------------------ | :------------ | :------------ | :------------ |
| **Device mode**     | -         | -         | -         |
| **Cloud mode**      | **Supported** | **Supported** | **Supported** |

Once you have confirmed that the source platform supports sending events to Marketo Static Lists, follow these steps:

1. From your [RudderStack dashboard](https://app.rudderstack.com/), add the source. Then, from the list of destinations, select **Marketo Static Lists**.
2. Assign a name to the destination and click **Continue**.

## Connection settings

To successfully configure Marketo Static Lists as a destination in RudderStack, you need to configure the following settings:

- **Munchkin Account ID**: Enter your Munchkin account ID. For more information on obtaining your Munchkin ID, refer to the <Link to="#faq">FAQ</Link> section below.
- **Client ID**: Enter your Marketo client ID.
- **Client Secret**: Enter the associated client secret for the above client ID.

<div class="infoBlock">
For more information on obtaining your Marketo client ID and secret, refer to the <Link to="#faq">FAQ</Link> section below.
</div>

> - **Static List ID**: Enter your Marketo static list ID. RudderStack uses this ID to add or remove leads from the specific list. For more information on obtaining your static list ID, refer to the <Link to="#faq">FAQ</Link> section below.

Reviewer: Can we add one more line here that this Static List Id is used as a fallback if Static List Id is provided in externalId

Author: Added a note addressing this suggestion.

Once you have confirmed that the source platform supports sending events to Marketo Static Lists, follow these steps:

1. From your [RudderStack dashboard](https://app.rudderstack.com/), add the source. Then, from the list of destinations, select **Marketo Static Lists**.
2. Assign a name to the destination and click **Continue**.

## Connection settings

To successfully configure Marketo Static Lists as a destination in RudderStack, you need to configure the following settings:

- **Munchkin Account ID**: Enter your Munchkin account ID. For more information on obtaining your Munchkin ID, refer to the <Link to="#faq">FAQ</Link> section below.
- **Client ID**: Enter your Marketo client ID.
- **Client Secret**: Enter the associated client secret for the above client ID.

<div class="infoBlock">
For more information on obtaining your Marketo client ID and secret, refer to the <Link to="#faq">FAQ</Link> section below.
</div>

- **Static List ID**: Enter your Marketo static list ID. RudderStack uses this ID to add or remove leads from the specific list. For more information on obtaining your static list ID, refer to the <Link to="#faq">FAQ</Link> section below.

<div class="infoBlock">

> RudderStack prioritizes the <code class="inline-code">marketoStaticListId</code> provided in the event's <code class="inline-code">externalId</code> over the <strong>Static List ID</strong> dashboard setting. If <code class="inline-code">marketoStaticListId</code> is absent in the payload, RudderStack looks for the static list ID specified in this setting.

Author: ```suggestion
RudderStack prioritizes the <code class="inline-code">marketoStaticListId</code> provided in the event's <code class="inline-code">externalId</code> over the <strong>Static List ID</strong> dashboard setting. If <code class="inline-code">marketoStaticListId</code> is absent in <code class="inline-code">externalId</code>, RudderStack looks for the static list ID specified in this setting.
```

Author: @anantjain45823 does this look fine?

Reviewer: Yes

> RudderStack prioritizes the <code class="inline-code">marketoStaticListId</code> provided in the event's <code class="inline-code">externalId</code> over the <strong>Static List ID</strong> dashboard setting. If <code class="inline-code">marketoStaticListId</code> is absent in <code class="inline-code">externalId</code>, RudderStack looks for the static list ID specified in this setting.

Reviewer 2: ```suggestion
RudderStack prioritizes the <code class="inline-code">marketoStaticListId</code> provided in the event's <code class="inline-code">externalId</code> over the <strong>Static List ID</strong> dashboard setting. If <code class="inline-code">marketoStaticListId</code> is absent in <code class="inline-code">externalId</code>, RudderStack looks for the static list ID in this setting as it is mandatory to send event data successfully to Marketo.
```

<img src="../../assets/event-stream-destinations/marketo-munchkin-id.png" alt="Marketo Munchkin ID" />

<div class="infoBlock">
For more information, refer to the <a href="https://nation.marketo.com/t5/knowledgebase/how-to-find-your-munchkin-id-for-a-marketo-instance/ta-p/248432">Marketo documentation</a>.
</div>

### How do I obtain the Marketo client ID and secret?

To set up the Marketo API service and obtain the client ID and secret associated with it, follow these steps:

1. Log into your Marketo instance and click the **Admin** tab.
2. Select **LaunchPoint**.

<img src="../../assets/event-stream-destinations/marketo-launchpoint.png" />

3. Here, you will able to see all the installed services used for connecting to Marketo.

<img src="../../assets/event-stream-destinations/marketo-services.png" />

4. To create a new service, click **New** > **New Service**.
5. Enter the **Display Name**. From the **Service** dropdown, select **Custom**.
6. Under **Settings**, enter the **Description** and select the **API Only User**, as shown. Finally, click **CREATE**.

<img src="../../assets/event-stream-destinations/marketo-services.png" />

<div class="warningBlock">

> Make sure the <strong>API Only User</strong> associated with the API service has the necessary permissions to create or update contacts as wellas the custom activities.

Reviewer 2: ```suggestion
Make sure the <strong>API Only User</strong> associated with the API service has the necessary permissions to create or update contacts as well as the custom activities.
```
